### PR TITLE
Update zenodo source of gridded weather data

### PR DESCRIPTION
**Pull request by brynpickering**
Fixes some review comments from @irm-codebase in #390.

Made a specific PR as they are changes that are worth considering in isolation, I believe. Deals with data units, mostly.

I'll deal with the rest of the review comments in #390 directly.

## Checklist

Any checks which are not relevant to the PR can be pre-checked by the PR creator. All others should be checked by the reviewer. You can add extra checklist items here if required by the PR.

- [ ] CHANGELOG updated
- [ ] Minimal workflow tests pass
- [ ] Tests added to cover contribution
- [ ] Documentation updated
- [x] Configuration schema updated

### CODE REVIEW

File: config/schema.yaml [hunk 1]
```diff
@@ -111,14 +111,10 @@ properties:
...
+            gridded-weather-data:
                 type: string
-                pattern: ^(https?|http?):\/\/.+
-                description: "Web address of gridded temperature data. Expecting a netCDF file with the coordinates [time, site], and the variables [temperature, lat, lon]."
-            gridded-10m-windspeed-data:
-                type: string
-                pattern: ^(https?|http?):\/\/.+
-                description: "Web address of gridded 10m wind speed data. Expecting a netCDF file with the coordinates [time, site], and the variables [temperature, lat, lon]."
+                pattern: ^(https?|http?):\/\/.+{data_var}.*\.(nc)$
+                description: "Web address of gridded weather data. Expecting a netCDF file with the coordinates [time, site], the variables [{data_type}, lat, lon], and the attribute 'unit'."
```
Comment (timtroendle) [inline config/schema.yaml:117]: Should be `[{data_var}, lat, lon]` I guess?

File: scripts/heat/unscaled_heat_profiles.py [hunk 2]
```diff
@@ -218,17 +222,14 @@ def get_daily_heat_demand(
 
     def heat_function(t, parameters):
```
Comment (timtroendle) [inline scripts/heat/unscaled_heat_profiles.py:223]: Type hints are missing everywhere but I understand they will be added in #390.

**Review by timtroendle — APPROVED** [review]
Looks good to me.

File: scripts/heat/unscaled_heat_profiles.py (hunk 2 above)
Comment (brynpickering) [reply scripts/heat/unscaled_heat_profiles.py:223]: Yes, and better variable naming

**Pull request merged**